### PR TITLE
Add ability to cancel a quote

### PR DESCRIPTION
**Pull request by marcofucci**
This adds the ability to reopen an order and cancel a quote.

A new endpoint POST `v3/order/<order-id>/quote/cancel` has been created.

The reason behind POSTing to cancel instead of DELETing the resource is that we do want the cancelled quote to still be referenced by the order so that it can be used to check its status.

We do not delete any quotes for the time being as they are going to be useful in absence of audit trail.

A cancelled quote can be replaced by a new one.

### CODE REVIEW

File: datahub/omis/order/test/test_models.py [hunk 1]
```diff
@@ -128,6 +133,58 @@ def test_without_committing(self):
         assert not Quote.objects.count()
 
 
+class TestReopen:
+    """Tests for when an order is reopened."""
+
+    def test_without_quote(self):
+        """
+        Test that if an order without quote is reopened, nothing happens as
+        the quote is already open.
```
Comment (reupen) [inline datahub/omis/order/test/test_models.py]: > the quote is already open
Is that right or should it say 'order is already open'?

Comment (marcofucci) [reply datahub/omis/order/test/test_models.py]: Nice catch!

**Comment by marcofucci** [comment]
@reupen is it a 👍  for you too? Just double-checking to be sure.

**Comment by marcofucci** [comment]
@reupen added some docs ;)

**Comment by reupen** [comment]
Looks good :)

**Comment by marcofucci** [comment]
need to wait for your bugfix before merging ;)

**Pull request merged**